web-api/tests/split_documents.rs

```diff
@@ -0,0 +1,448 @@
+    send_assert(
+        client,
+        client
+            .put(ingestion_url.join("/documents/_candidates")?)
+            .json(&json!({ "documents": ids }))
+            .build()?,
+        StatusCode::NO_CONTENT,
+        false,
+    )
+    .await;
+    Ok(())
+}
+
+async fn get_properties(
+    client: &Client,
+    ingestion_url: &Url,
+    id: &str,
+) -> Result<Map<String, Value>, Error> {
+    let mut url = ingestion_url.clone();
+    url.path_segments_mut()
+        .unwrap()
+        .extend(["documents", id, "properties"]);
+
+    #[derive(Deserialize)]
+    struct Response {
+        properties: Map<String, Value>,
+    }
+
+    Ok(
+        send_assert_json::<Response>(client, client.get(url).build()?, StatusCode::OK, false)
+            .await
+            .properties,
+    )
+}
+
+fn json_map(
+    pairs: impl IntoIterator<Item = (impl Into<String>, impl Into<Value>)>,
+) -> Map<String, Value> {
+    pairs
+        .into_iter()
+        .map(|(k, v)| (k.into(), v.into()))
+        .collect()
+}
+
+async fn ingest(client: &Client, ingestion_url: &Url) -> Result<(), Error> {
+    send_assert(
+        client,
+        client
+            .post(ingestion_url.join("/documents")?)
+            .json(&json!({
+                "documents": [
+                    {
+                        "id": "d1",
+                        "snippet": "The duck is blue. The truck is yellow. But birds are birds.",
+                        "split": true,
+                        "properties": {
+                            "foo": "filter-a",
+                        }
+                    },
+                    {
+                        "id": "d2",
+                        "snippet": "Cars with wheels. This is a blue sentence. Trains on tracks. ",
+                        "split": true,
+                        "properties": {
+                            "foo": "filter-a",
+                        }
+                    },
+                ]
+            }))
+            .build()?,
+        StatusCode::CREATED,
+        false,
+    )
+    .await;
+
+    Ok(())
+}
+
+#[test]
+fn test_split_documents_for_semantic_search() {
+    test_two_apps::<Ingestion, Personalization, _>(
+        UNCHANGED_CONFIG,
+        with_dev_options(),
+        |client, ingestion_url, personalization_url, _| async move {
+            let query = |text, count| query(&client, &personalization_url, text, count, None);
+            ingest(&client, &ingestion_url).await?;
+
+            let res = query("cars", 2).await?;
+            assert_eq!(res.id_ranking(), vec!["d2"]);
+            assert_eq!(res.documents[0].splits, [0, 2].into_iter().collect());
+
+            let res = query("color", 3).await?;
+            assert_eq!(res.id_ranking(), vec!["d1", "d2"]);
+            assert_eq!(res.documents[0].splits, [0, 1].into_iter().collect());
+            assert_eq!(res.documents[1].splits, [1].into_iter().collect());
```

Comment:
:art: the set can directly be collected like this. similarly below.
```suggestion
            assert_eq!(res.documents[0].splits, [0, 1].into());
            assert_eq!(res.documents[1].splits, [1].into());
```